### PR TITLE
Moved and revised the default operator article

### PR DESCRIPTION
**Pull request by pkulikov**
Contributes to #10741
Supported by dotnet/samples#1132

### CODE REVIEW

**Comment by rpetrusha** [comment]
Closing and reopening to begin new build after dotnet/samples#1132 was merged.

**Comment by rpetrusha** [comment]
Code did not appear in the most recent build. Closing and reopening to begin new build.

File: docs/csharp/language-reference/keywords/default.md [hunk 1]
```diff
@@ -14,8 +13,9 @@ ms.assetid: 14c48aaa-7d35-4058-a1a4-f53353050579
 # default (C# Reference)
 
 The `default` keyword can be used in two ways:
-- The [`switch` statement](switch.md) to specify the default label.
-- A [default value expression](../../programming-guide/statements-expressions-operators/default-value-expressions.md) to produce the default value of a type.
+
+- To specify the default label in the [`switch` statement](switch.md).
+- As the [default operator](../operators/default.md) to produce the default value of a type.
```
Comment (rpetrusha) [inline docs/csharp/language-reference/keywords/default.md]: Should this also mention the default literal?

File: docs/csharp/language-reference/operators/default.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+title: "default operator - C# reference"
+ms.custom: seodec18
+description: "Use default operator to produce the default value of a type"
+ms.date: 08/01/2019
+helpviewer_keywords: 
+  - "default keyword [C#]"
+---
+# default operator (C# reference)
+
+The `default` operator produces the [default value](../keywords/default-values-table.md) of a type. An argument of the `default` operator must be the name of a type or a type parameter.
```
Comment (rpetrusha) [inline docs/csharp/language-reference/operators/default.md]:
```suggestion
The `default` operator produces the [default value](../keywords/default-values-table.md) of a type. The argument to the `default` operator must be the name of a type or a type parameter.
```

File: docs/csharp/language-reference/operators/default.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+[!code-csharp-interactive[default of T](~/samples/csharp/language-reference/operators/DefaultOperator.cs#WithOperand)]
+
+You also use the `default` keyword as the default case label within the [`switch` statement](../keywords/switch.md).
+
+## default literal
+
+Beginning with C# 7.1, you can use the `default` literal to produce the default value of a type when the compiler can infer the expression type. The `default` literal expression produces the same value as the `default(T)` expression where `T` is the inferred type. You can use the `default` literal in any of the following cases:
+
+- assignment or initialization of a variable
```
Comment (rpetrusha) [inline docs/csharp/language-reference/operators/default.md]:
```suggestion
- In the assignment or initialization of a variable.
```

File: docs/csharp/language-reference/operators/default.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+- declaration of the default value for an optional method parameter
```
Comment (rpetrusha) [inline docs/csharp/language-reference/operators/default.md]:
```suggestion
- In the declaration of the default value for an optional method parameter.
```

File: docs/csharp/language-reference/operators/default.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+- providing the argument value in a method call
```
Comment (rpetrusha) [inline docs/csharp/language-reference/operators/default.md]:
```suggestion
- In an argument to a method call.
```

File: docs/csharp/language-reference/operators/default.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+- in a `return` statement or as an expression in an expression-bodied member
```
Comment (rpetrusha) [inline docs/csharp/language-reference/operators/default.md]:
```suggestion
- In a `return` statement or as an expression in an expression-bodied member.
```

**Review by rpetrusha — CHANGES_REQUESTED** [review]
Thanks for doing this, @pkulikov. This looks good. I've left a few suggestions and one question for you to consider.

**Comment by pkulikov** [comment]
@rpetrusha thanks for the review and suggestions. I've addressed your feedback; please take a look at the latest commit.

**Comment by rpetrusha** [comment]
This looks very good, @pkulikov; thanks for the additional changes. I'll merge your PR now.

**Review by rpetrusha — APPROVED** [review]

**Pull request merged**